scalyr_agent/builtin_monitors/kubernetes_events_monitor.py

```diff
@@ -0,0 +1,478 @@
+                # has expired, so make sure to only check for a new leader if the interval has expired
+                if last_check + self._leader_check_interval <= current_time:
+                    last_check = current_time
+                    # check if we are the leader
+                    if not self._is_leader( k8s ):
+                        #if not, then sleep and try again
+                        global_log.log( scalyr_logging.DEBUG_LEVEL_0, "Leader is %s" % (str(self._current_leader)) )
+                        self._sleep_but_awaken_if_stopped(self._leader_check_interval)
+                        continue
+
+                    global_log.log( scalyr_logging.DEBUG_LEVEL_0, "Leader is %s" % (str(self._current_leader)) )
+                try:
+                    # start streaming events
+                    lines = k8s.stream_events( last_event=last_event )
+
+                    json = {}
+                    for line in lines:
+                        try:
+                            json = json_lib.parse(line)
+                        except Exception, e:
+                            global_log.warning( "Error parsing event json: %s, %s, %s" % (line, str(e), traceback.format_exc() ) )
+                            continue
+
+                        obj = json.get( "object", JsonObject() )
+
+                        # check to see if the resource version we are using has expired
+                        if json.get( 'type', '' ) == 'ERROR':
+                            if obj.get( 'kind', '' ) == 'Status' and obj.get( 'reason', '' ) == 'Expired':
+                                # if so, reset to None to get all events, we'll skip over the already seen
+                                # ones later
+                                last_event = None
+                                global_log.log( scalyr_logging.DEBUG_LEVEL_1, "K8S resource expired" )
+                                continue
+
+                        # resource version hasn't expired, so update it to the most recently seen version
+                        last_event = last_resource
+
+                        metadata = obj.get( "metadata", JsonObject() )
+
+                        # skip any events with resourceVersions higher than ones we've already seen
+                        resource_version = metadata.get_int( "resourceVersion", None, none_if_missing=True )
+                        if resource_version and resource_version <= last_resource:
+                            global_log.log( scalyr_logging.DEBUG_LEVEL_2, "Skipping older resource events" )
+                            continue
+
+                        last_resource = resource_version
+                        last_event = resource_version
+
+                        # see if this event is about an object we are interested in
+                        involved = obj.get( "involvedObject", JsonObject())
```

Comment:
How about write a method called `parse_involved` (or something like that) that returns `kind, name, namespace` and use it here.  This method has a bit too many lines and I think that gets in the place of readability.

Reply:
Done.